Assets/Fungus/Scripts/Editor/FlowchartWindow.cs

```diff
@@ -531,129 +583,343 @@ protected virtual void DrawBlockPopup(Flowchart flowchart)
+                        menu.AddItem(new GUIContent ("Cut"), false, () => Cut(flowchart));
+                        menu.AddItem(new GUIContent ("Duplicate"), false, () => Duplicate(flowchart));
+                        menu.AddItem(new GUIContent ("Delete"), false, () => AddToDeleteList(blockList));
+                    }
+                    else
+                    {
+                        DeselectAll();
+
+                        menu.AddItem(new GUIContent("Add Block"), false, () => CreateBlock(flowchart, mousePosition / flowchart.Zoom - flowchart.ScrollPos));
+
+                        if (copyList.Count > 0)
+                        {
+                            menu.AddItem(new GUIContent("Paste"), false, () => Paste(flowchart, mousePosition));
+                        }
+                        else
+                        {
+                            menu.AddDisabledItem(new GUIContent("Paste"));
+                        }
+                    }
+
+                    var menuRect = new Rect();
+                    menuRect.position = new Vector2(mousePosition.x, mousePosition.y - 12f);
+                    menu.DropDown(menuRect);
+                    Event.current.Use();               
+                }
+                break;
             }
-            
-            // The center of the Flowchart depends on the block positions and window dimensions, so we calculate it 
-            // here in the FlowchartWindow class and store it on the Flowchart object for use later.
-            if (flowchart != null && blocks.Length > 0)
+
+            // Selection box
+            selectionBox.size = Vector2.zero;
+            selectionBox.position = -Vector2.one;
+            startSelectionBoxPosition = selectionBox.position;
+        }
+
+        protected override void OnScrollWheel(Event e)
+        {
+            if (selectionBox.size == Vector2.zero)
             {
-                CalcFlowchartCenter(flowchart, blocks);
+                Vector2 zoomCenter;
+                zoomCenter.x = e.mousePosition.x / flowchart.Zoom / position.width;
+                zoomCenter.y = e.mousePosition.y / flowchart.Zoom / position.height;
+                zoomCenter *= flowchart.Zoom;
+
+                DoZoom(flowchart, -Event.current.delta.y * 0.01f, zoomCenter);
+                e.Use();
             }
+        }
+        
+        protected virtual void DrawFlowchartView()
+        {
+            var e = Event.current;
+
+            // Calc rect for script view
+            Rect scriptViewRect = new Rect(0, 0, this.position.width / flowchart.Zoom, this.position.height / flowchart.Zoom);
 
-            // Draw connections
-            foreach (var block in blocks)
-            {
-                DrawConnections(flowchart, block, false);
-            }
-            foreach (var block in blocks)
-            {
-                DrawConnections(flowchart, block, true);
```

Comment:
Yep, that was it. It looked weird when a non-highlighted block overlapped in front of a highlighted one. It seemed easier to do it this way rather than maintain two separate lists for highlighted / non-highlighted.